router/h2/src/main/scala/io/buoyant/router/h2/ClassifierFilter.scala

```diff
@@ -33,18 +37,30 @@ object ClassifierFilter {
   }
 }
 
-class ClassifierFilter(classifier: ResponseClassifier) extends SimpleFilter[Request, Response] {
+class ClassifierFilter(classifier: H2StreamClassifier) extends SimpleFilter[Request, Response] {
 
   private[this] val successClass = classifier.andThen(_.fractionalSuccess).lift
 
   def apply(req: Request, svc: Service[Request, Response]): Future[Response] = {
-    svc(req).map { rep =>
-      successClass(ReqRep(req, Return(rep))) match {
-        case Some(success) =>
+    svc(req).map {
+      case rep if rep.stream.isEmpty =>
+        successClass(H2ReqRep(req, Return(rep), None)).foreach { success =>
           rep.headers.set(ClassifierFilter.SuccessClassHeader, success.toString)
-        case None =>
-      }
-      rep
+        }
+        rep
+      case rep =>
+        val stream = rep.stream.onFrame {
+          case Return(f: Trailers) =>
+            successClass(H2ReqRep(req, Return(rep), Some(Return(f)))).foreach { success =>
+              f.set(ClassifierFilter.SuccessClassHeader, success.toString)
+            }
+          case Return(f) if f.isEnd =>
+            successClass(H2ReqRep(req, Return(rep), Some(Return(f)))).foreach { success =>
+              rep.headers.set(ClassifierFilter.SuccessClassHeader, success.toString)
```

Comment:
I'm wondering if we can add a method that is to `onFrame` as `map` is to `foreach`...